File: Leetcode/1935 - Maximum Number of Words You Can Type/solution.cpp
int canBeTypedWords(string text, string brokenLetters) {
    vector<bool> brokenKeys(26, false);
    for (char c: brokenLetters) brokenKeys[c - 'a'] = true;

    int count = 0;
    bool brokenKeyUsed = false;
    for (char c: text) {
        if (c == ' ') {
            if (!brokenKeyUsed) count++;
            brokenKeyUsed = false;
        } else if (brokenKeys[c - 'a']) {
            brokenKeyUsed = true;
        }
    }
    if (!brokenKeyUsed) count++;

    return count;
}

File: Leetcode/1935 - Maximum Number of Words You Can Type/solution.js
var canBeTypedWords = function(text, brokenLetters) {
    const brokenKeys = Array(26).fill(false);
    for (const c of brokenLetters) {
        brokenKeys[c.charCodeAt(0) - 97] = true;
    }

    let count = 0;
    let brokenKeyUsed = false;
    for (const c of text) {
        if (c === ' ') {
            if (!brokenKeyUsed) count++;
            brokenKeyUsed = false;
        } else if (brokenKeys[c.charCodeAt(0) - 97]) {
            brokenKeyUsed = true;
        }
    }
    if (!brokenKeyUsed) count++;

    return count;
};